### PR TITLE
[Main Course] - Timeseries algo

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

post for Timeseries algo

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-7-18/1626574877658-2.png" width="500" />
+  <figcaption>outliers</figcaption>
+</figure>
+
+通过肉眼，我们可以发现异常值在第6000个观测值附近。然而那一块并没有出现极值。因此，可以说对于这种周期性较强的时间序列，极值可能也是周期的一部分。我们需要识别的是某个与其他周期不同的小区间。
+
+## 5 Matrixprofile算法
+
+那么，如何识别出周期层面的异常区间呢？我们可能会想，我们先确定一个区间，然后看看这个区间和其他同样长度的区间长的是否不同。
```
Comment (BulletTech2021) [inline docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md:53]: ```长的是否不同``` -》```长得是否不同```

File: docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+我们用刚开始瞧一瞧的时间序列为例。通过人眼识别，我们发现异常值在第5500个观测值附近，我们看看用matrixprofile将会表现如何？
+
+通过计算，matrixprofile告诉我们mp值最大的子序列的起始点为5500，整个序列的mp值序列如下图所示：
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-7-18/1626576279441-4.png" width="500" />
+  <figcaption>mp series</figcaption>
+</figure>
+
+可以看到，这个子序列相较别的序列，在距离上有明显的不同。我们成功找到了和人眼识别完全相同的异常序列，我们可以简单的认为异常值在这个子序列的中点。
```
Comment (BulletTech2021) [inline docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md:85]: 简单地

File: docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-7-18/1626574877658-2.png" width="500" />
+  <figcaption>outliers</figcaption>
+</figure>
+
+通过肉眼，我们可以发现异常值在第6000个观测值附近。然而那一块并没有出现极值。因此，可以说对于这种周期性较强的时间序列，极值可能也是周期的一部分。我们需要识别的是某个与其他周期不同的小区间。
+
+## 5 Matrixprofile算法
+
+那么，如何识别出周期层面的异常区间呢？我们可能会想，我们先确定一个区间，然后看看这个区间和其他同样长度的区间长的是否不同。  
+题目好心的提示我们有这样一个包：matrixprofile。
```
Comment (BulletTech2021) [inline docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md:54]: 好心地

**Review by BulletTech2021 — APPROVED** [review]
挺直白的，🆗

File: docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+</figure>
+
+通过肉眼，我们可以发现异常值在第6000个观测值附近。然而那一块并没有出现极值。因此，可以说对于这种周期性较强的时间序列，极值可能也是周期的一部分。我们需要识别的是某个与其他周期不同的小区间。
+
+## 5 Matrixprofile算法
+
+那么，如何识别出周期层面的异常区间呢？我们可能会想，我们先确定一个区间，然后看看这个区间和其他同样长度的区间长得是否不同。  
+题目好心地提示我们有这样一个包：matrixprofile。
+
+[matrixprofile](https://towardsdatascience.com/introduction-to-matrix-profiles-5568f3375d90)算法由加州大学河滨分校的研究者于2016年提出。它的想法比较简单直白，即给定窗口长度，计算这个子序列与其他同样长度的子序列的距离，其中，与最相似序列(距离最小的序列)的距离为这个子序列的mp值。通过滑动窗口，我们可以计算出整个序列的mp值。其中，较大的mp值可以认为，此处的子序列与和它最相似的序列距离都比较大，因此该子序列有更大的可能是异常值。
```
Comment (jz263) [inline docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md:56]: 这里是不是可以简单描述一下window size可以怎么取，例如我们参考了傅里叶变换

File: docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    作者：Void，发布于2021-07-16，阅读时间：约5分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s/3JlO0eO95SBqhHH4EHPypw)
+
+## 1 引言
+
+事情的起因是有朋友告诉我最近有KDD Cup 2021的比赛。为了凑个热闹，也为了刷点经验，我们准备合伙参加(当个炮灰)。  
+有三道赛题，时间序列异常检测、图相关的和智慧城市。看上去最正常的[时间序列异常检测](https://compete.hexagon-ml.com/practice/competition/39/)当仁不让的成为了我们的选择。
+
+## 2 题目要求
+
+竞赛要求我们检测多个时间序列(25+250个)中的异常点。每个时间序列有且仅有一个异常点。题目给出了异常点所在的区间，要求我们给出异常点所在的位置。
```
Comment (jz263) [inline docs/Main_Course/Data_Analytics/2021-07-16-Timeseries-algo.md:17]: 可以考虑提一下竞赛分为两期，第一期phase1 有 25个数据集用于调试算法，phase2 有250个 test数据集作为比赛成绩

**Pull request merged**